src/brevitas/nn/target/flexml.py

```diff
@@ -0,0 +1,98 @@
+from typing import Union
+
+import torch
+from torch import Tensor
+from torch import nn
+
+import brevitas.nn as qnn
+from brevitas.function.ops_ste import ceil_ste
+from brevitas.function.ops import max_int
+from brevitas.nn.mixin import QuantLayerMixin
+from brevitas.core.utils import StatelessBuffer
+
+from brevitas.quant import Int8ActPerTensorFixedPoint
+from brevitas.quant import Uint8ActPerTensorFixedPoint
+from brevitas.quant_tensor import QuantTensor
+
+
+class FlexMLQuantLeakyReLU(nn.Module):
+
+    def __init__(
+            self,
+            negative_slope,
+            alpha_quant=qnn.QuantIdentity(Uint8ActPerTensorFixedPoint, bit_width=16),
+            input_quant=qnn.QuantIdentity(Int8ActPerTensorFixedPoint, bit_width=16, scaling_stats_momentum = None),
+            output_quant=qnn.QuantIdentity(Int8ActPerTensorFixedPoint, return_quant_tensor=True)):
+        super(FlexMLQuantLeakyReLU, self).__init__()
+        self.alpha_quant = alpha_quant
+        self.input_quant = input_quant
+        self.output_quant = output_quant
+        self.negative_slope = StatelessBuffer(torch.tensor(negative_slope))
+
+    def forward(self, inp):
+        quant_inp = self.input_quant(inp)
+        quant_alpha = self.alpha_quant(self.negative_slope())
+        quant_alpha_out = self.input_quant(quant_inp * quant_alpha)
+        out = torch.max(quant_inp, quant_alpha_out)
+        out = self.output_quant(out)
+        return out
+
+
+class FlexMLQuantAvgPool2d(QuantLayerMixin, nn.AvgPool2d):
+
+    class Int16QuantAvgPoolDivQuant(Int8ActPerTensorFixedPoint):
+        bit_width = 8  # div_factor is a 16b in the backend so it could go up to 16b in principle
+        scaling_stats_op = 'max'
+        float_to_int_impl_type = 'round'  # round to nearest ties to even
+
+    def __init__(
+            self,
+            kernel_size,
+            stride = None,
+            padding = 0,
+            ceil_mode = False,
+            div_quant = Int16QuantAvgPoolDivQuant,
+            return_quant_tensor=True) -> None:
+        nn.AvgPool2d.__init__(
+            self,
+            kernel_size=kernel_size,
+            stride=stride,
```

Comment:
I noticed that the strides are set to `[7, 7]` we need `[1, 1]`. Not sure where this is coming from.

Reply:
The logic for converting an adaptive avg pool to an avg pool assuming the output size divides the input size is (source https://stackoverflow.com/a/63603993/16744139):
stride = input_size // output_size
kernel_size = input_size - (output_size - 1) * stride
padding = 0

Reply:
Ah I see, then in this case when kernel=stride=[H, W] would setting stride=[1, 1] be equivalent? I am just asking because I notice Vitis AI Quantizer setting the stride to [1, 1] for this case.

Reply:
I have added a flag with default True to set the stride to 1 whenever the average pool is global (i.e. across all features).